opensearch-operator/api/v1/opensearch_types.go

```diff
@@ -107,10 +107,17 @@ type ConfMgmt struct {
 }
 
 type MonitoringConfig struct {
-	Enable               bool   `json:"enable,omitempty"`
-	MonitoringUserSecret string `json:"monitoringUserSecret,omitempty"`
-	ScrapeInterval       string `json:"scrapeInterval,omitempty"`
-	PluginURL            string `json:"pluginUrl,omitempty"`
+	Enable               bool                `json:"enable,omitempty"`
+	MonitoringUserSecret string              `json:"monitoringUserSecret,omitempty"`
+	ScrapeInterval       string              `json:"scrapeInterval,omitempty"`
+	PluginURL            string              `json:"pluginUrl,omitempty"`
+	Scheme               string              `json:"scheme,omitempty"`
```

Comment:
I don't think scheme should be configurable:
1. At the moment the operator does not support non-tls-protected clusters so `https` is the only possible value.
2. Even if later on the cluster supported it then the scheme should be implicitly calculated by the operator based on if TLS is activated. The user shouldn't have to care about it.

Reply:
Removed